testplan/common/utils/match.py

```diff
@@ -296,22 +538,54 @@ def get_between(self, mark1=None, mark2=None):
         :return: The content between mark1 and mark2.
         :rtype: ``str``
         """
-        if mark1 is not None and mark2 is not None:
-            if self.marks[mark1] >= self.marks[mark2]:
-                raise ValueError(
-                    'Mark "{}" must be present before mark "{}"'.format(
-                        mark1, mark2
+        try:
+            log_handle = None
+
+            if mark1 is not None and mark2 is not None:
+                if self.strategy.invalid_position_order(
```

Comment:
@kn-ms I am thinking the invalid_position_order should be maintained instead of the compare since we might have multiple files and will not be doing the comparison only on the position but on the files inode as well.